src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+        {
+            // Page 41 documentation PN5180A0XX-C3.pdf
+            if ((configuration.Length > 252) || (configuration.Length % 6 != 0) || (configuration.Length == 0))
+            {
+                throw new ArgumentException($"Configuration buffer size must be a multiple of 5 and no larger than 195 bytes");
+            }
+
+            Span<byte> rfConfig = stackalloc byte[1 + configuration.Length];
+            rfConfig[0] = (byte)Command.UPDATE_RF_CONFIG;
+            configuration.CopyTo(rfConfig.Slice(1));
+
+            try
+            {
+                SpiWrite(rfConfig);
+            }
+            catch (TimeoutException)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// True to disable the Radio Frequence collision avoidance according to ISO/IEC 18092
+        /// False to use Active Communication mode according to ISO/IEC 18092
+        /// </summary>
+        public RadioFrequenceCollision RadioFrequenceCollision { get; set; } = RadioFrequenceCollision.Normal;
+
+        /// <summary>
+        /// Get or set the radio frequence field. True for on, false for off
+        /// </summary>
+        public bool RadioFrequenceField
+        {
+            get
+            {
+                Span<byte> status = stackalloc byte[4];
+                try
+                {
+                    SpiReadRegister(Register.RF_STATUS, status);
+                }
+                catch (TimeoutException)
+                {
+                    return false;
+                }
+
+                return (status[2] & 0b1000_0000) == 0b1000_0000;
+            }
+            set
+            {
+                SetRadioFrequence(value);
+            }
+        }
+
+        /// <summary>
+        /// Get the radio frenquence status
+        /// </summary>
+        /// <returns>The radio frequence status</returns>
+        public RadioFrequenceStatus GetRadioFrequenceStatus()
+        {
+            Span<byte> status = stackalloc byte[4];
+            try
+            {
+                SpiReadRegister(Register.RF_STATUS, status);
+            }
+            catch (TimeoutException)
+            {
+                return RadioFrequenceStatus.Error;
+            }
+
+            return (RadioFrequenceStatus)((status[2] >> 1) & 0x07);
+        }
+
+        /// <summary>
+        /// Is the external field activated?
+        /// </summary>
+        /// <returns>True if active, false if not</returns>
+        public bool IsRadioFrequenceFieldExternal()
+        {
+            Span<byte> status = stackalloc byte[4];
+            try
+            {
+                SpiReadRegister(Register.RF_STATUS, status);
+            }
+            catch (TimeoutException)
+            {
+                return false;
+            }
+
+            return (status[2] & 0b0000_0100) == 0b0000_0100;
```

Comment:
any chance you could cast this to enum and use HasFlag or at minimum comment what this bit is? disregard if this was ported

Reply:
the status for this specific field is just 1 bit out of a 4 bytes configuration. So I'm making the test only once like this. It's clearly documented in the documentation of the device